authorization-codegen/build.gradle

```diff
@@ -0,0 +1,18 @@
+plugins {
+    id 'java-library'
+    id 'io.deephaven.project.register'
+
+    // TODO: fix this hack and invoke the protoc during the build via docker; maybe convert to a replicate task
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
+}
+
+description 'Deephaven Authorization Code Generator'
+
+dependencies {
+     api project(':engine-api')
+
+     Classpaths.inheritGrpcPlatform(project, 'api')
+     api 'io.grpc:grpc-services'
+
+    implementation 'com.squareup:javapoet:1.9.0'
+}
```

Comment:
I see there aren't any tasks that generate the files automatically. I think that's _fine_; it's a manually process right now?

I think we have three options:

1. Leave as-is, need to know to manually code-gen as necessary.
2. Add gradle tasks that can check, and then run to regenerate.
3. Add test that can generate and compare.

Either way, I think we need at least a comment here or quick README file here that explains process.

Reply:
i think this step is probably waiting on ~me, to help with changing how the docker container is built that runs protoc

Reply:
I've spawned #3133. I am adding to the readme how to regenerate the files.